### PR TITLE
Implement network switching

### PR DESCRIPTION
**Pull request by mmjee**
# Pull Request

Resolves [AR-3411](https://team-1624093970686.atlassian.net/browse/AR-3411).

## Blocking dependencies

N/A.

## Changes

Implemented support for switching networks, and adding the network automatically to the wallet if not present.

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -171,12 +171,44 @@ export class StorageProvider {
...
+        if (e.code === 4902) {
+          await this.provider.provider.request({
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: hexChainID,
+                chainName: 'Arcana',
+
+                rpcUrls: [chainIdToRPCURL.get(this.chainId)],
+                // blockExplorerUrls: ['https://testnet.bscscan.com'],
```
Comment (SauravKanchan) [inline src/index.ts:195]: can you remove this commented code?

Or add our explorer endpoint

**Review by SauravKanchan — CHANGES_REQUESTED** [review]

**Comment by mmjee** [comment]
Alright, I'll add a new mapping.

File: src/index.ts (hunk 1 above)
Comment (mmjee) [reply src/index.ts]: Addressed.

**Review by SauravKanchan — APPROVED** [review]

File: src/constant.js [hunk 2]
```diff
@@ -5,3 +5,7 @@ export var chainIdToGateway = new Map([
     [40405, "https://gateway001-testnet.arcana.network/api/v1/"],
     [40404, "https://gateway-testnet.arcana.network/api/v1/"]
 ]);
+export var chainIdToRPCURL = new Map([
```
[the remark this answers is not in the document]
Comment (mmjee) [reply src/constant.js:10]: We can follow up later on that I suppose.

**Pull request merged**